src/app/clusters/boolean-state-configuration-server/boolean-state-configuration-server.cpp

```diff
@@ -274,30 +274,46 @@ CHIP_ERROR SetAllEnabledAlarmsActive(EndpointId ep)
     VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == AlarmsEnabled::Get(ep, &alarmsEnabled),
                         CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute));
 
-    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == AlarmsActive::Set(ep, alarmsEnabled), CHIP_IM_GLOBAL_STATUS(Failure));
-    emitAlarmsStateChangedEvent(ep);
+    if (alarmsEnabled.HasAny())
+    {
+        VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == AlarmsActive::Set(ep, alarmsEnabled), CHIP_IM_GLOBAL_STATUS(Failure));
```

Comment:
For a follow-up, let's stop using EMBER_ZCL_xxx and let's use InteractionModel::Status instead

Reply:
Sure, I can try addressing that in a follow up PR 👍🏼